app/models/comment.php

```diff
@@ -20,17 +20,116 @@ public static function getComments($offset, $limit, $id)
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment 
                         WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($id));
-       
+
+        foreach ($rows as $row) {
+            $favorites = array();
+            $user_favorites = $db->rows("SELECT * FROM favorites WHERE comment_id = ?", array($row['id']));
+            $total_favorites = Comment::countFavorites($row['id']);
+
+            foreach ($user_favorites as $v) {
+                $favorites[] = $v['user_id'];
+            }
+            $row['favorites'] = $favorites; 
+            $row['total_favorites'] = $total_favorites;
+            $comments[] = new Comment($row);
+        }
+        return $comments;
+    }
+
+    public static function total($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ?", array($thread_id));
+    }
+
+    public function edit($comment_id)
+    {
+        if (!$this->body) {
+            throw new RecordNotFoundException('Record Not Found');
+        }
+        
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->query('UPDATE comment SET body = ?, created = NOW()
+                    WHERE id = ?', array($this->body, $comment_id));
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public static function getComment($comment_id)
+    {
+        $user = array();
```

Comment:
Please change name to `$comments` maybe? the objects are from "Comment" class. Its not user